File: Easy/3274_Check_if_Two_Chessboard_Squares_Have_the_Same_Color/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    bool squareIsWhite(string coordinates) {
        int row = coordinates[0] - '0';
        int col = coordinates[1] - 'a';

        return (row + col) % 2 == 0;
    }

    bool checkTwoChessboards(string coordinate1, string coordinate2) {
        return squareIsWhite(coordinate1) == squareIsWhite(coordinate2);
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    string coordinate1_1 = "a1", coordinate1_2 = "c3";
    bool result1 = solution.checkTwoChessboards(coordinate1_1, coordinate1_2);
    cout << (result1 ? "true" : "false") << endl;
    
    // test cases 2
    string coordinate2_1 = "a1", coordinate2_2 = "h3";
    bool result2 = solution.checkTwoChessboards(coordinate2_1, coordinate2_2);
    cout << (result2 ? "true" : "false") << endl;

    return 0;
}

File: Easy/3274_Check_if_Two_Chessboard_Squares_Have_the_Same_Color/Solution.java


public class Solution {
    public boolean squareIsWhite(String coordinates) {
        int col = coordinates.charAt(0) - 'a' + 1;
        int row = coordinates.charAt(1) - '0';

        return (row + col) % 2 == 0;
    }

    public boolean checkTwoChessboards(String coordinate1, String coordinate2) {
        return squareIsWhite(coordinate1) == squareIsWhite(coordinate2);
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        String coordinate1_1 = "a1";
        String coordinate1_2 = "c3";
        System.out.println(solution.checkTwoChessboards(coordinate1_1, coordinate1_2));
        
        // test cases 2
        String coordinate2_1 = "a1";
        String coordinate2_2 = "h3";
        System.out.println(solution.checkTwoChessboards(coordinate2_1, coordinate2_2));
        
    }
}